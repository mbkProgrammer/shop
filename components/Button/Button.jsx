import { useTheme } from '@emotion/react';
import styled from '@emotion/styled';
import PropTypes from 'prop-types';

const Button = ({
  varaint, children, onClick, size = 'medium',
}) => {
  const theme = useTheme();

  const Button = styled.button`
  border-radius: 5px;
  color: ${theme.colors.primary};
  width: fit-content;
  font-size: 19px;
  font-weight: 400;
  border: none;
  border-radius: 5px;
  margin: 5px;
  transition: 0.3s;
  &.btn--text {
    color: ${theme.colors.text};
    background: none;
  }
  &.btn--outlined {
    border: 1px solid;
    background: ${theme.colors.secondary};
  }
  &.btn--outlined:hover {
    border: 1px solid;
    background: ${theme.colors.primary};
    color: ${theme.colors.secondary};
  }

  &.btn--contained {
    background: ${theme.colors.primary};
    border: 1px solid ${theme.colors.primary};
    color: #fafafa;
  }
  &.btn--contained:hover, &.btn--text:hover {
    background-image: linear-gradient(rgba(0, 0, 0, 0.2) 0 0);
  }
  &.btn--big {
    padding: 8px 74px;
    min-width: fit-content;
    max-width: 100%;
  }

  &.btn--small {
    padding: 8px 24px;
  }

  &.btn--medium {
    padding: 8px 44px;
  }

  `;

  return <Button className={`btn--${varaint} btn--${size} `} onClick={onClick}>{children}</Button>;
};
Button.propTypes = {
  varaint: PropTypes.oneOf(['text', 'outlined', 'contained']),
  size: PropTypes.oneOf(['small', 'medium', 'big']),
};

Button.defaultProps = {
  varaint: 'text',
  size: 'medium',
};

export default Button;
