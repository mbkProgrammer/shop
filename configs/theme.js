const theme = {
  colors: {
    primary: '#56B280',
    secondary: '#F5FBF8',
    text: '#222322',
    background: '#EAF6F0',
  },
  typography: {
    body: {
      fontWeight: '300',
      fontSize: '1rem',
    },
    h1: {
      fontWeight: '700',
      fontSize: '2.4rem',
    },
    h2: {
      fontWeight: '700',
      fontSize: '2.2rem',
    },
    h3: {
      fontWeight: '700',
      fontSize: '2rem',
    },
    h4: {
      fontWeight: '600',
      fontSize: '1.8rem',
    },
    h5: {
      fontWeight: '600',
      fontSize: '1.6rem',
    },
    h6: {
      fontWeight: '500',
      fontSize: '1.4rem',
    },
  },
};

export default theme;
