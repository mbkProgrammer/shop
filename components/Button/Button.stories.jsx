import React from 'react';
import Button from './Button';

export default { title: 'components/Button' };
export const TextSmall = () => (<Button varaint="text" size="small">Button</Button>);
export const TextMedium = () => (<Button varaint="text">Button</Button>);
export const TextBig = () => (<Button varaint="text" size="big">Button</Button>);
export const OutlinedSmall = () => (<Button varaint="outlined" size="small">Button</Button>);
export const OutlinedMedium = () => (<Button varaint="outlined">Button</Button>);
export const OutlinedBig = () => (<Button varaint="outlined" size="big">Button</Button>);
export const ContainedSmall = () => (<Button varaint="contained" size="small">Button</Button>);
export const ContainedMedium = () => (<Button varaint="contained">Button</Button>);
export const ContainedBig = () => (<Button varaint="contained" size="big">Button</Button>);
