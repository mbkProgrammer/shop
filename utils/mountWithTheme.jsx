import { ThemeProvider } from '@emotion/react';
import renderer from 'react-test-renderer';
import theme from '../configs/theme';

const mountWithTheme = (children) => renderer.create(
  <ThemeProvider theme={theme}>
    {children}
  </ThemeProvider>,
);

export default mountWithTheme;
