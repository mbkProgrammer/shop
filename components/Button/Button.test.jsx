import { ThemeProvider } from '@emotion/react';
import React from 'react';
import renderer from 'react-test-renderer';
import { cleanup, fireEvent, render } from '@testing-library/react';
import Button from './Button';
import theme from '../../configs/theme';
import mountWithTheme from '../../utils/mountWithTheme';

test('Button smaple', () => {
  const component = mountWithTheme(
    <Button>Facebook</Button>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Button smaple contained', () => {
  const component = mountWithTheme(
    <Button varaint="contained">Facebook</Button>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Button smaple onClick', () => {
  let variable = 1;
  const { getByText } = render(
    <ThemeProvider theme={theme}>
      <Button varaint="contained" onClick={() => { variable += 1; }}>Facebook</Button>
    </ThemeProvider>,
  );

  fireEvent.click(getByText(/Facebook/));
  expect(variable).toBe(2);
});
