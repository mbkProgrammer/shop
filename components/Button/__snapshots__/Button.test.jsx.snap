// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button smaple 1`] = `
<button
  className="btn--text btn--medium  css-zr5tfn"
>
  Facebook
</button>
`;

exports[`Button smaple big 1`] = `
<button
  className="btn--text btn--big  css-zr5tfn"
>
  Facebook
</button>
`;

exports[`Button smaple contained 1`] = `
<button
  className="btn--contained btn--medium  css-zr5tfn"
>
  Facebook
</button>
`;
